### ISSUE TITLE
Regent: arrays are immutable if initialised from function

### ISSUE THREAD

**Issue by rupanshusoi** [StanfordLegion/legion#728]
The following
```
import "regent"

local C = regentlib.c

task t()
        var c : int[1]
        c[0] = 1
        return c
end

task main()
        var a = t()
        a[0] = 2
        C.printf("%d\n", a[0])
end
regentlib.start(main)
```
outputs `1`. It's as if the compiler just ignores the `a[0] = 2` statement. Even if I create a new array in `main` and initialise it like so
```
var a = t()
var b = a
b[0] = 2
```
the output is identical. 
Interestingly, the bug vanishes by just changing `task t()` to `terra t()`.

**Comment by elliottslaughter** [comment]
I see the problem. Working on this now.

**Comment by elliottslaughter** [comment]
This should be fixed in 5fe63912072e64b18150cde8607a5dfa8356c8e6. Note that it can take several minutes for the Github mirror to sync with Gitlab.
